### ISSUE TITLE
BEM Diagnostic Incorrect Reportings

### ISSUE THREAD

**Issue by zeshanshani** [Box-Of-Hats/Bem-VSCode-Extension#9]
Hey! I have been enjoying this plugin a lot. Thanks a lot for putting so much effort in it! 💯 

I have encountered one strange issue where the BEM diagnostics reports incorrectly when used in PHP file (see: https://s.zeshanahmed.com/a_2341ECB8.png). 

It even gives false warnings in a pure PHP code: https://s.zeshanahmed.com/a_2341ED9D.png

Can you check what could be wrong and why it's giving such false warnings?

Thanks a lot,
Zeshan

**Comment by Box-Of-Hats** [comment]
Thanks for the feedback :) 

I'll take a look into this issue later today and see what I can do

**Comment by Box-Of-Hats** [comment]
I've been unable to replicate the issue, even after typing the examples in your screenshots.

Neither of these lines raise any diagnostic issues for me:
```
function zawp_generate_btn_classes( $classes = '', $btn_color = 'green', $outline = false)
```
```
<?php if ( have_rows( '_ctas') ): ?>
```

Could you provide the code that is causing the issue?

**Comment by zeshanshani** [comment]
@Box-Of-Hats that's strange. Are you checking it on `.php` files? It's basically raising errors on most of the PHP variables and functions as well (see: https://s.zeshanahmed.com/a_23451DC3.png)

e.g., here's another piece of code in a PHP file that I showed in the screenshot:

```
// If it has background image, add custom class.
if ( isset( $adv_styling['bg_image'] ) && $adv_styling['bg_image'] ) {
  $classes_arr[] = 'section--bg-image';
}

// If it has background image, add custom class.
if ( isset( $adv_styling['dark_bg'] ) && $adv_styling['dark_bg'] ) {
  $classes_arr[] = 'section--bg-dark';
}
```

And yes, the settings are as you mentioned.

**Comment by Box-Of-Hats** [comment]
I'm testing it on a `.php` file using the contents you provided but I'm still not getting any issues.

![image](https://user-images.githubusercontent.com/9083502/66077440-e8abcf00-e557-11e9-8a27-693a4a3c40b8.png)

Although I've not been able to replicate the issue, I found a potential bug in the code which *may* have been causing it.
I've released a new version (0.7.3) with [this fix](https://github.com/Box-Of-Hats/Bem-VSCode-Extension/commit/717ce2ce3dc077deea69b20d8c06d847b4357ccc)

If this still doesn't fix it for you, I recommend changing your settings to use `"bemHelper.classNameCase": "any",` as that will prevent BEM Helper from flagging up any classname casing issues it finds

**Comment by Box-Of-Hats** [comment]
I've finally managed to reproduce and find the source of this bug and I'll be working on a fix shortly

**Comment by zeshanshani** [comment]
Hey mate, sorry I forgot to respond to your earlier message. 

I'll try the fix asap and will let you know if it helps. 

And good to hear that you were able to reproduce the bug. Thanks for working on it! 🎉

**Comment by Box-Of-Hats** [comment]
Fix has been made and deployed in version 0.7.4.

Let me know if this has fixed it for you or if you're still having issues

[commit](https://github.com/Box-Of-Hats/Bem-VSCode-Extension/commit/52c99ad564678f7c89c89f6925f97ce6e6d3a439)

**Comment by zeshanshani** [comment]
@Box-Of-Hats that works great. Thanks a lot! :)

However, I have encountered a few other issues. I'll open them up in a separate issue. Let's make this extension perfect for BEM users! 💯 🥇
